src/plugins/vis_type_vega/public/data_model/vega_parser.ts

```diff
@@ -22,33 +22,64 @@ import * as vega from 'vega-lib';
 import * as vegaLite from 'vega-lite';
 import schemaParser from 'vega-schema-url-parser';
 import versionCompare from 'compare-versions';
-import { EsQueryParser } from './es_query_parser';
+// @ts-ignore
 import hjson from 'hjson';
+import { VISUALIZATION_COLORS } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { EsQueryParser } from './es_query_parser';
 import { Utils } from './utils';
 import { EmsFileParser } from './ems_file_parser';
 import { UrlParser } from './url_parser';
-import { VISUALIZATION_COLORS } from '@elastic/eui';
-import { i18n } from '@kbn/i18n';
+import { SearchAPI } from './search_api';
+import { TimeCache } from './time_cache';
+import { IServiceSettings } from '../../../maps_legacy/public';
+import { Bool, Data, VegaSpec } from './types';
 
 // Set default single color to match other Kibana visualizations
-const defaultColor = VISUALIZATION_COLORS[0];
+const defaultColor: string = VISUALIZATION_COLORS[0];
 
-const DEFAULT_SCHEMA = 'https://vega.github.io/schema/vega/v3.0.json';
+const DEFAULT_SCHEMA: string = 'https://vega.github.io/schema/vega/v3.0.json';
 
-const locToDirMap = {
+const locToDirMap: any = {
   left: 'row-reverse',
   right: 'row',
   top: 'column-reverse',
   bottom: 'column',
 };
 
 // If there is no "%type%" parameter, use this parser
-const DEFAULT_PARSER = 'elasticsearch';
+const DEFAULT_PARSER: string = 'elasticsearch';
 
 export class VegaParser {
-  constructor(spec, searchAPI, timeCache, filters, serviceSettings) {
+  spec: VegaSpec | string;
+  hideWarnings: boolean;
+  error?: string;
+  warnings: string[];
+  _urlParsers: { [key: string]: any };
+  isVegaLite?: boolean;
+  useHover?: boolean;
+  _config?: any;
+  useMap?: boolean;
+  renderer?: string;
+  tooltips?: object;
+  mapConfig?: object;
+  vlspec?: any;
```

Comment:
Seems like this is a `VegaSpec` as well?

Reply:
👍